python/orca/src/bigdl/orca/common.py

```diff
@@ -183,79 +180,95 @@ def init_orca_context(cluster_mode="local", cores=2, memory="2g", num_nodes=1,
            Default to be False and in this case the Ray cluster would be launched lazily when
            Ray is involved in Project Orca.
     :param kwargs: The extra keyword arguments used for creating SparkContext and
-           launching Ray if any.
-
+           launching Ray if any. 
+           For k8s-client mode, you should specify master as ${SPARK_MASTER} (k8s://https://localhost:8443)
+           and set container_image to ${K8s_SPARK_IMAGE}.
     :return: An instance of SparkContext.
     """
     print("Initializing orca context")
     import atexit
     atexit.register(stop_orca_context)
-    cluster_mode = cluster_mode.lower()
+    from pyspark import SparkContext, SparkConf
+    import warnings
     spark_args = {}
     for key in ["conf", "spark_log_level", "redirect_spark_log"]:
         if key in kwargs:
             spark_args[key] = kwargs[key]
-    if cluster_mode == "spark-submit":
+    if cluster_mode is not None:
+        cluster_mode = cluster_mode.lower()
+    activate_sc = SparkContext._active_spark_context is not None
+    if activate_sc:
+        if cluster_mode is not None:
+            warnings.warn("Use an existing SparkContext, " +
+                          "cluster_mode is determined by the existing SparkContext", Warning)
+        conf = SparkConf()
```

Comment:
> is it necessary? As you dig into the code, https://github.com/intel-analytics/BigDL/blob/branch-2.0/python/dllib/src/bigdl/dllib/nncontext.py#L678 it will create a SparkConf by default, yes?

Yes, it could create a sparkconf object same as system default, which is necessary in https://github.com/sgwhat/BigDL/blob/755b04f3ce023aab0adae5c9292bbe7f64ad7f81/python/orca/src/bigdl/orca/common.py#L206.